application/actions/meetings.go

```diff
@@ -1,11 +1,183 @@
 package actions
 
 import (
+	"context"
+	"errors"
+	"github.com/gobuffalo/buffalo"
+	"github.com/gobuffalo/buffalo/render"
 	"github.com/silinternational/wecarry-api/api"
 	"github.com/silinternational/wecarry-api/models"
+	"time"
 )
 
-func convertMeeting(meeting models.Meeting) api.Meeting {
+// swagger:operation GET /events Events ListEvents
+//
+// gets a list of meetings
+//
+// ---
+// responses:
+//   '200':
+//     description: meetings list
+//     schema:
+//       "$ref": "#/definitions/Meetings"
+func meetingsList(c buffalo.Context) error {
+	cUser := models.CurrentUser(c)
+	tx := models.Tx(c)
+	meetings := models.Meetings{}
+	if err := meetings.FindOnOrAfterDate(tx, time.Now().UTC()); err != nil {
+		return reportError(c, api.NewAppError(err, api.ErrorMeetingsGet, api.CategoryInternal))
+	}
+
+	output, err := convertMeetings(c, meetings, cUser)
+	if err != nil {
+		return reportError(c, api.NewAppError(err, api.ErrorMeetingsConvert, api.CategoryInternal))
+	}
+
+	return c.Render(200, render.JSON(output))
+}
+
+// converts list of model.Meeting into list of api.Meeting
+func convertMeetings(ctx context.Context, meetings []models.Meeting, user models.User) ([]api.Meeting, error) {
+	output := make([]api.Meeting, len(meetings))
+
+	for i, m := range meetings {
+		var err error
+		output[i], err = convertMeeting(ctx, m, user)
+		if err != nil {
+			return []api.Meeting{}, err
+		}
+	}
+
+	return output, nil
+}
+
+// converts a model.Meeting into api.Meeting
+func convertMeeting(ctx context.Context, meeting models.Meeting, user models.User) (api.Meeting, error) {
+	output := convertMeetingAbridged(meeting)
+
```

Comment:
My last refactor on `convertRequest` did one Load up front and then mostly just type conversions after that. You could try that here, but I'm not convinced it is better. The one tangible benefit was that some of the helper functions didn't have an error return.